### PR TITLE
timer: change id values

### PR DESCRIPTION
**Pull request by tlauda**
Refactors timer a little bit, to make better usage
of timer id values. Right now every timer has also irq
field, which is used to identify which interrupt line
to enable/disable. We don't longer need timer id to represent
also timer irq, so let's set custom ids and simplify
things.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

### CODE REVIEW

File: src/arch/xtensa/drivers/timer.c [hunk 1]
```diff
@@ -16,32 +16,19 @@ struct timer_data {
 	void *arg2;
 };
```
Comment (dbaluta) [inline src/arch/xtensa/drivers/timer.c]: @tlauda please make this change also for imx: src/platform/imx8/include/platform/drivers/timer.h

Working on multiple platforms comes with the responsability of keeping the code consistent and a change should cover all existing platforms.

File: src/arch/xtensa/drivers/timer.c [hunk 2]
```diff
@@ -16,32 +16,19 @@ struct timer_data {
...
 
-static struct timer_data xtimer[3] = {};
+static struct timer_data xtimer[TIMER3] = {};
 
 void timer_64_handler(void *arg)
```
Comment (dbaluta) [inline src/arch/xtensa/drivers/timer.c]: TIME3 here doesn't sound good. Something like TIMER_COUNT would make better sense.

File: src/arch/xtensa/drivers/timer.c (hunk 1 above)
Comment (tlauda) [reply src/arch/xtensa/drivers/timer.c]: @dbaluta It's done for IMX. I'm always covering all of the platforms.

Comment (dbaluta) [reply src/arch/xtensa/drivers/timer.c]: @tlauda Sorry, didn't notice that :)

**Comment by tlauda** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
Jenkins CI invalid failures. Now merged.

**Pull request merged**